fedimint-client/src/db.rs

```diff
@@ -441,6 +460,63 @@ pub fn get_core_client_database_migrations() -> BTreeMap<DatabaseVersion, CoreMi
         })
     });
 
+    // Migration to add outcome_time to OperationLogEntry
+    migrations.insert(DatabaseVersion(1), |mut ctx| {
+        Box::pin(async move {
+            let mut dbtx = ctx.dbtx();
+
+            // Read all OperationLogEntries using V0 format
+            let operation_logs = dbtx
+                .find_by_prefix(&OperationLogKeyPrefixV0)
+                .await
+                .collect::<Vec<_>>()
+                .await;
+
+            // Build a map from operation_id -> max_time of inactive state
+            let mut op_id_max_time = BTreeMap::new();
+
+            // Process inactive states
+            let inactive_states = dbtx
+                .find_by_prefix(&InactiveStateKeyPrefixBytes)
+                .await
+                .collect::<Vec<_>>()
+                .await;
+
+            for (state, meta) in inactive_states {
+                let entry = op_id_max_time
+                    .entry(state.operation_id)
+                    .or_insert(meta.created_at);
+                *entry = (*entry).max(meta.created_at).max(meta.exited_at);
+            }
+
+            // Migrate each V0 operation log entry to the new format
+            for (op_key_v0, log_entry_v0) in operation_logs {
+                let new_entry = OperationLogEntry {
+                    operation_module_kind: log_entry_v0.operation_module_kind,
+                    meta: log_entry_v0.meta,
+                    outcome_time: log_entry_v0.outcome.as_ref().map(|_| {
+                        // If we found state times, use the max, otherwise use current time
+                        op_id_max_time
+                            .get(&op_key_v0.operation_id)
+                            .copied()
+                            .unwrap_or_else(fedimint_core::time::now)
```

Comment:
Shouldn't it be `None` for still-active operations? Makes me think: we'd need to filter out operations with active states.

We can ignore it and accept slightly off data I guess.

Reply:
as_ref().map, so it will be none if there was no outcome